server/src/handlers/http/ingest.rs

```diff
@@ -49,6 +51,32 @@ pub async fn ingest(req: HttpRequest, body: Bytes) -> Result<HttpResponse, PostE
         let stream_name = stream_name.to_str().unwrap().to_owned();
         create_stream_if_not_exists(&stream_name).await?;
 
+        if first_event_at_empty(&stream_name) {
```

Comment:
IMHO we shouldn't do this in the ingestion flow. we can do this at the `get_stats` API call handler.

Reply:
Changes done.